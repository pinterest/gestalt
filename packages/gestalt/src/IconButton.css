:root {
  --lightGray: #efefef;
}

.button {
  composes: block from "./Layout.css";
  composes: noBorder from "./Borders.css";
  composes: p0 from "./Whitespace.css";
  composes: pointer from "./Cursor.css";
  background: transparent;
}

.button:focus {
  outline: 0;
}
