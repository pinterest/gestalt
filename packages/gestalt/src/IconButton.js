// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import styles from './IconButton.css';
import icons from './icons/index.js';
import Pog from './Pog.js';

type Props = {|
  accessibilityExpanded?: boolean,
  accessibilityHaspopup?: boolean,
  accessibilityLabel: string,
  bgColor?:
    | 'transparent'
    | 'transparentDarkGray'
    | 'gray'
    | 'lightGray'
    | 'white',
  iconColor?: 'gray' | 'darkGray' | 'red' | 'blue' | 'white',
  icon: $Keys<typeof icons>,
  onClick?: ({ event: SyntheticMouseEvent<> }) => void,
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
|};

type State = {|
  active: boolean,
  focused: boolean,
  hovered: boolean,
|};

export default class IconButton extends React.Component<Props, State> {
  static propTypes = {
    accessibilityExpanded: PropTypes.bool,
    accessibilityHaspopup: PropTypes.bool,
    accessibilityLabel: PropTypes.string.isRequired,
    bgColor: PropTypes.oneOf([
      'transparent',
      'transparentDarkGray',
      'gray',
      'lightGray',
      'white',
    ]),
    icon: PropTypes.oneOf(Object.keys(icons)).isRequired,
    iconColor: PropTypes.oneOf(['gray', 'darkGray', 'red', 'blue', 'white']),
    onClick: PropTypes.func,
    size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
  };

  state = {
    active: false,
    focused: false,
    hovered: false,
  };

  handleBlur = () => this.setState({ focused: false });

  handleFocus = () => this.setState({ focused: true });

  handleMouseDown = () => this.setState({ active: true });

  handleMouseEnter = () => this.setState({ hovered: true });

  handleMouseLeave = () => this.setState({ active: false, hovered: false });

  handleMouseUp = () => this.setState({ active: false });

  render() {
    const {
      accessibilityExpanded,
      accessibilityHaspopup,
      accessibilityLabel,
      bgColor,
      iconColor,
      icon,
      size,
      onClick,
    } = this.props;

    const { active, focused, hovered } = this.state;

    return (
      <button
        aria-expanded={accessibilityExpanded}
        aria-haspopup={accessibilityHaspopup}
        aria-label={accessibilityLabel}
        className={styles.button}
        onBlur={this.handleBlur}
        onClick={event => onClick && onClick({ event })}
        onFocus={this.handleFocus}
        onMouseDown={this.handleMouseDown}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
        onMouseUp={this.handleMouseUp}
        type="button"
      >
        <Pog
          active={active}
          bgColor={bgColor}
          focused={focused}
          hovered={hovered}
          iconColor={iconColor}
          icon={icon}
          size={size}
        />
      </button>
    );
  }
}
