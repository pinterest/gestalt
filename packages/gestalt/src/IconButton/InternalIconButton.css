.button {
  composes: block from "../Layout.css";
  composes: noBorder from "../Borders.css";
  composes: p0 from "../Whitespace.css";
  background-color: var(--color-transparent);
}

.parentButton {
  composes: noBorder from "../Borders.css";
  composes: p0 from "../Whitespace.css";
  composes: paddingX0 from "../boxWhitespace.css";
  composes: paddingY0 from "../boxWhitespace.css";
  background-color: var(--color-transparent);
}

.label {
  max-width: 16ch;
  padding-top: var(--space-200);
}

.parentButton:focus {
  outline: 0;
}

.enabled {
  composes: pointer from "../Cursor.css";
}

.disabled {
  cursor: default;
  opacity: 0.5;
}

.disabledVr {
  cursor: default;
}

.activeText div {
  /* Ensure the disabled color is used */
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--sema-color-pressed-text-default) !important;
}

.hoverText div {
  /* Ensure the disabled color is used */
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--sema-color-hover-text-default) !important;
}
