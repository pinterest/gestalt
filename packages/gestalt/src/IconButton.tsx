import { forwardRef, useImperativeHandle, useRef } from 'react';
import InternalIconButton from './IconButton/InternalIconButton';
import icons from './icons/index';
import { Indexable } from './zIndex';

type Props = {
  /**
   * Label for screen readers to announce IconButton. See the [Accessibility](https://gestalt.pinterest.systems/web/iconbutton#ARIA-attributes) guidelines for details on proper usage.
   */
  accessibilityLabel: string;
  /**
   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by IconButton so that screen reader users can identify the relationship between elements. See the [Accessibility](https://gestalt.pinterest.systems/web/iconbutton#ARIA-attributes) guidelines for details on proper usage.
   */
  accessibilityControls?: string;
  /**
   * Indicates that IconButton hides or exposes collapsible components and expose whether they are currently expanded or collapsed. See the [Accessibility](https://gestalt.pinterest.systems/web/iconbutton#ARIA-attributes) guidelines for details on proper usage.
   */
  accessibilityExpanded?: boolean;
  /**
   * Indicates that a component controls the appearance of interactive popup elements, such as menu or dialog. See the [Accessibility](https://gestalt.pinterest.systems/web/iconbutton#ARIA-attributes) guidelines for details on proper usage.
   */
  accessibilityHaspopup?: boolean;
  /**
   * Indicates whether this component displays a menu, such as Dropdown, or a dialog, like Popover, Modal or ModalAlert. See the [Accessibility](https://gestalt.pinterest.systems/web/iconbutton#ARIA-attributes) guidelines for details on proper usage.
   */
  accessibilityPopupRole?: 'menu' | 'dialog';
  /**
   * Indicates whether this component is hosted in a light or dark container.
   * Used for improving focus ring color contrast.
   */
  focusColor?: 'lightBackground' | 'darkBackground';
  /**
   * Primary colors to apply to the IconButton background.
   */
  bgColor?:
    | 'transparent'
    | 'transparentDarkGray'
    | 'gray'
    | 'lightGray'
    | 'washLight'
    | 'white'
    | 'red';
  /**
   * Defines a new icon different from the built-in Gestalt icons.
   */
  dangerouslySetSvgPath?: {
    __path: string;
  };
  /**
   * Available for testing purposes, if needed. Consider [better queries](https://testing-library.com/docs/queries/about/#priority) before using this prop.
   */
  dataTestId?: string;
  /**
   * When disabled, IconButton looks inactive and cannot be interacted with.
   */
  disabled?: boolean;
  /**
   * Icon displayed in IconButton to convey the behavior of the component. Refer to the [iconography](/foundations/iconography/library#Search-icon-library) guidelines regarding the available icon options.
   */
  icon?: keyof typeof icons;
  /**
   * Primary color to apply to the [Icon](/web/icon). See [icon color](https://gestalt.pinterest.systems/web/iconbutton#Icon-color) variant to learn more.
   */
  iconColor?: 'gray' | 'darkGray' | 'red' | 'white' | 'brandPrimary' | 'light' | 'dark';
  /**
   * Visible label for the IconButton. Only visible in XL size IconButtons. See the [label](https://gestalt.pinterest.systems/web/iconbutton#Label) variant to learn more.
   */
  label?: string;
  /**
   * The name attribute specifies the name of the button element. The name attribute is used to reference form-data after the form has been submitted and for [testing](https://testing-library.com/docs/queries/about/#priority).
   */
  name?: string;
  /**
   * Callback fired when the component is clicked, pressed or tapped.
   */
  onClick?: (arg1: {
    event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>;
  }) => void;
  /**
   * Sets a padding for the IconButton. See the [size](#Size) variant to learn more.
   */
  padding?: 1 | 2 | 3 | 4 | 5;
  /**
   * Ref that is forwarded to the underlying button element.
   */
  // eslint-disable-next-line react/no-unused-prop-types
  ref?: HTMLButtonElement;
  /**
   * Toggles between binary states: on/off, selected/unselected, open/closed. See the [selected](https://gestalt.pinterest.systems/web/iconbutton#Selected-state) variant to learn more.
   */
  selected?: boolean;
  /**
   * The maximum height and width of IconButton. See the [size](https://gestalt.pinterest.systems/web/iconbutton#Size) variant to learn more.
   */
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  /**
   * Removes IconButton from sequential keyboard navigation to improve accessibility. See the [Accessibility](https://gestalt.pinterest.systems/web/iconbutton#Keyboard-interaction) guidelines for details on proper usage.
   */
  tabIndex?: -1 | 0;
  /**
   * Adds a [Tooltip](/web/tooltip) on hover/focus of the IconButton.
   */
  tooltip?: {
    accessibilityLabel?: string;
    inline?: boolean;
    idealDirection?: 'up' | 'right' | 'down' | 'left';
    text: string;
    zIndex?: Indexable;
  };
  /**
   * Use "submit" if IconButton is used within or associated with a form.
   */
  type?: 'submit' | 'button';
};

/**
 * [IconButton](https://gestalt.pinterest.systems/web/iconbutton) allows users to take actions and make choices with a single click or tap. IconButtons use icons instead of text to convey available actions on a screen. IconButton is typically found in forms, dialogs and toolbars.
 Some buttons are specialized for particular tasks, such as navigation or presenting menus.
 *
 * ![IconButton light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/IconButton.spec.ts-snapshots/IconButton-chromium-darwin.png)
 * ![IconButton dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/IconButton-dark.spec.ts-snapshots/IconButton-dark-chromium-darwin.png)
 *
 */

const IconButtonWithForwardRef = forwardRef<HTMLButtonElement, Props>(function IconButton(
  {
    accessibilityControls,
    accessibilityExpanded,
    accessibilityHaspopup,
    accessibilityLabel,
    accessibilityPopupRole,
    bgColor,
    focusColor = 'lightBackground',
    dangerouslySetSvgPath,
    dataTestId,
    disabled,
    icon,
    iconColor,
    label,
    name,
    onClick,
    padding,
    selected,
    size = 'lg',
    tabIndex = 0,
    tooltip,
    type,
  }: Props,
  ref,
) {
  const innerRef = useRef<null | HTMLButtonElement>(null);
  // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
  // that renders <IconButton ref={inputRef} /> to call inputRef.current.focus()
  // @ts-expect-error - TS2322 - Type 'HTMLButtonElement | null' is not assignable to type 'HTMLButtonElement'.
  useImperativeHandle(ref, () => innerRef.current);

  return (
    <InternalIconButton
      ref={innerRef}
      accessibilityControls={accessibilityControls}
      accessibilityExpanded={accessibilityExpanded}
      accessibilityHaspopup={accessibilityHaspopup}
      accessibilityLabel={accessibilityLabel}
      accessibilityPopupRole={accessibilityPopupRole}
      bgColor={bgColor}
      dangerouslySetSvgPath={dangerouslySetSvgPath}
      dataTestId={dataTestId}
      disabled={disabled}
      focusColor={focusColor}
      icon={icon}
      iconColor={iconColor}
      label={label}
      name={name}
      onClick={onClick}
      padding={padding}
      selected={selected}
      size={size}
      tabIndex={tabIndex}
      tooltip={tooltip}
      type={type === 'submit' ? 'submit' : 'button'}
    />
  );
});

IconButtonWithForwardRef.displayName = 'IconButton';

export default IconButtonWithForwardRef;
