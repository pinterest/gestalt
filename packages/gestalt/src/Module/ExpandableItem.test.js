// @flow strict
import renderer from 'react-test-renderer';
import ModuleExpandableItem from './ExpandableItem.js';
import IconButton from '../IconButton.js';

describe('ModuleExpandableItem', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          isCollapsed
          type="info"
          onModuleClicked={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with icon', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          isCollapsed
          icon="lock"
          iconAccessibilityLabel="lock icon label"
          type="info"
          onModuleClicked={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with badge', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          badge={{ text: 'Try it out!' }}
          id="uniqueTestID"
          isCollapsed
          onModuleClicked={() => {}}
          title="test title"
          type="info"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with icon button', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          isCollapsed
          iconButton={
            <IconButton
              bgColor="lightGray"
              icon="question-mark"
              iconColor="darkGray"
              accessibilityLabel="Get help"
              size="xs"
              onClick={() => {}}
            />
          }
          type="info"
          onModuleClicked={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with summary', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          isCollapsed
          summary={['summary1', 'summary2', 'summary3']}
          type="info"
          onModuleClicked={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with error', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          iconAccessibilityLabel="there is an error"
          isCollapsed
          type="error"
          onModuleClicked={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with children', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          isCollapsed
          type="info"
          onModuleClicked={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with when expanded', () => {
    const tree = renderer
      .create(
        <ModuleExpandableItem
          id="uniqueTestID"
          accessibilityExpandLabel="click to expand"
          accessibilityCollapseLabel="click to collapse"
          title="test title"
          isCollapsed={false}
          summary={['summary1', 'summary2', 'summary3']}
          type="info"
          onModuleClicked={() => {}}
        >
          <div>test children</div>
        </ModuleExpandableItem>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
