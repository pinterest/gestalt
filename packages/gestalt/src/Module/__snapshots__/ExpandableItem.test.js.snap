// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleExpandableItem renders correctly 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with badge 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="box marginStart2 relative"
                    style={
                      Object {
                        "top": "1px",
                      }
                    }
                  >
                    <div
                      className="Badge middle infoBase"
                    >
                      <div
                        className="Flex rowGap1 columnGap1 xsDirectionRow xsItemsCenter"
                      >
                        <div
                          className="FlexItem"
                        >
                          <div
                            className="box xsDisplayInlineBlock"
                            style={
                              Object {
                                "marginTop": "2px",
                              }
                            }
                          >
                            Try it out!
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with children 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with error 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <svg
                    aria-hidden={null}
                    aria-label="there is an error"
                    className="icon errorIcon iconBlock"
                    height={16}
                    role="img"
                    viewBox="0 0 24 24"
                    width={16}
                  >
                    <path
                      d="test-file-stub"
                    />
                  </svg>
                </div>
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 errorText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with icon 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <svg
                    aria-hidden={null}
                    aria-label="lock icon label"
                    className="icon defaultIcon iconBlock"
                    height={16}
                    role="img"
                    viewBox="0 0 24 24"
                    width={16}
                  >
                    <path
                      d="test-file-stub"
                    />
                  </svg>
                </div>
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with icon button 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <button
                    aria-label="Get help"
                    className="parentButton"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      className="button tapTransition enabled"
                    >
                      <div
                        className="pog lightGray"
                        style={
                          Object {
                            "height": 24,
                            "width": 24,
                          }
                        }
                      >
                        <svg
                          aria-hidden={true}
                          aria-label=""
                          className="icon defaultIcon iconBlock"
                          height={12}
                          role="img"
                          viewBox="0 0 24 24"
                          width={12}
                        >
                          <path
                            d="test-file-stub"
                          />
                        </svg>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with summary 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="click to expand"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol6"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
              </div>
            </div>
            <div
              className="box marginStart6 xsCol6"
            >
              <div
                className="Flex rowGap0 columnGap2 xsDirectionColumn"
              >
                <div
                  className="FlexItem"
                >
                  <div
                    className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="summary1"
                  >
                    summary1
                  </div>
                </div>
                <div
                  className="FlexItem"
                >
                  <div
                    className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="summary2"
                  >
                    summary2
                  </div>
                </div>
                <div
                  className="FlexItem"
                >
                  <div
                    className="Text fontSize200 defaultText alignStart breakWord fontWeightNormal lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="summary3"
                  >
                    summary3
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleExpandableItem renders correctly with when expanded 1`] = `
<div
  className="box paddingX6 paddingY6"
>
  <div
    className="Flex rowGap0 columnGap6 xsDirectionColumn"
  >
    <div
      className="FlexItem"
    >
      <div
        aria-controls="uniqueTestID"
        aria-disabled={false}
        aria-expanded={true}
        aria-label="click to collapse"
        className="hideOutline tapTransition rounding0 accessibilityOutline fullWidth pointer"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="Flex rowGap0 columnGap0 xsDirectionRow"
        >
          <div
            className="box flexGrow marginEnd6 xsDisplayFlex xsItemsBaseline"
          >
            <div
              className="box xsCol12"
            >
              <div
                className="Flex rowGap2 columnGap0 xsDirectionRow xsItemsCenter"
              >
                <div
                  className="FlexItem"
                  style={
                    Object {
                      "minWidth": 0,
                    }
                  }
                >
                  <div
                    className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                    style={
                      Object {
                        "WebkitLineClamp": 1,
                      }
                    }
                    title="test title"
                  >
                    test title
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="box paddingX1 paddingY1 xsDisplayFlex xsItemsCenter"
            id="uniqueTestID"
            style={
              Object {
                "maxHeight": 24,
              }
            }
          >
            <svg
              aria-hidden={null}
              aria-label="click to collapse"
              className="icon defaultIcon iconBlock"
              height="12"
              role="img"
              viewBox="0 0 24 24"
              width="12"
            >
              <path
                d="test-file-stub"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="FlexItem"
    >
      <div>
        test children
      </div>
    </div>
  </div>
</div>
`;
